internal/expr/prepare.go

```diff
@@ -60,17 +60,27 @@ func starCount(fns []fullName) int {
 	return s
 }
 
-// prepareInput checks that the input expression corresponds to a known type.
-func prepareInput(ti typeNameToInfo, p *inputPart) (typeMember, error) {
-	info, ok := ti[p.sourceType.prefix]
+type typeNameToInfo map[string]typeInfo
+
+func (ti typeNameToInfo) lookupInfo(typeName string) (typeInfo, error) {
```

Comment:
[Nitpick] I think the naming of the function could be improved by keeping it simple and changing it to `lookup` because the type system has the rest of the information.

Another problem with the spelling right now is that you are really looking up a typeName, not the info, in the same way that you look up a word in a dictionary and not the meaning.

Reply:
Fair enough, I've changed it

Reply:
> Another problem with the spelling right now is that you are really looking up a typeName, not the info, in the same way that you look up a word in a dictionary and not the meaning.

This logic doesn't necessarily lead to a more clear API. When we go for an address book, are we looking for a person or for an address?  Phonebook, person or phone?  Etc.

So the answer is that, as usual, it depends.

With that said, it's great to see that there's interest in producing good APIs with sensible naming. Thanks for that.

Reply:
Following our discussion this morning @letFunny has done the changes to remove the method on `typeNameToInfo` and keep simply the map lookup, its looking a lot better.